### PR TITLE
Bug 2034493: *: Use --v=2 logging to drop client-side throttling noise

### PR DESCRIPTION
**Pull request by wking**
We've been using `--v=5` since 88c222c954 (#448).  But it leads to large quantities of [noise from client-side throttling][1], and that throttling is `V(3)`:

```console
$ grep -n . vendor/k8s.io/client-go/rest/request.go | grep -B7 -A6 '^597:'
589:    switch {
590:    case len(retryInfo) > 0:
591:            message = fmt.Sprintf("Waited for %v, %s - request: %s:%s", latency, retryInfo, r.verb, r.URL().String())
592:    default:
593:            message = fmt.Sprintf("Waited for %v due to client-side throttling, not priority and fairness, request: %s:%s", latency, r.verb, r.URL().String())
594:    }
596:    if latency > longThrottleLatency {
597:            klog.V(3).Info(message)
598:    }
599:    if latency > extraLongThrottleLatency {
600:            // If the rate limiter latency is very high, the log message should be printed at a higher log level,
601:            // but we use a throttled logger to prevent spamming.
602:            globalThrottledLogger.Infof("%s", message)
603:    }
```

Auditing:

```console
$ git --no-pager grep 'klog.V([3-5])' vendor
```

I don't see much that I'd miss.  I liked having `vendor/github.com/openshift/library-go/pkg/verify`'s logging back when we made that pivot, but now that that code is old and stable, I'm ok losing it.

I've shifted the `lib/` stuff down to `V(2)` using:

```console
$ sed -i 's/klog[.]V([3-5])/klog.V(2)/g' $(git grep -l klog.V lib)
```

It's mostly hotloop-detection since 40d0a4e401 (#561, [rhbz#1879184][2]), and isn't all that noisy since 05e1af7fba (#628, [rhbz#1984414][3]).

I've shifted the pkg/ stuff down to `V(2)` using:

```console
$ sed -i 's/klog[.]V([3-5])/klog.V(2)/g' $(git grep -l klog.V pkg)
```

It's mostly fairly core stuff, and [low-noise except for the per-manifest `Running sync for ...` and `Done syncing for ...` messages][1].  The per-manifest messages can be useful to identify where the CVO is in the sync cycle, so I'm keeping them for now.  We may be able to punt them up to higher levels if we get an alternative mechanism for identifying sync cycle positions (e.g. events for task-node completion or task failure).

[1]: https://bugzilla.redhat.com/show_bug.cgi?id=2034493#c1
[2]: https://bugzilla.redhat.com/show_bug.cgi?id=1879184
[3]: https://bugzilla.redhat.com/show_bug.cgi?id=1984414

### CODE REVIEW

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Comment by wking** [comment]
/hold

I want to peak at the CI results before this lands.

**Comment by wking** [comment]
[Three][1] [rounds][2] [of][3] `msg=Error: error creating/updating Private DNS Zone...The gateway did not receive a response from 'Microsoft.Network' within the specified time period.`.

/retest

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1478433982222700544#1:build-log.txt%3A49
[2]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-operator/1478433982319169536#1:build-log.txt%3A49
[3]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade/1478433982361112576#1:build-log.txt%3A69

**Comment by wking** [comment]
Looking at [the e2e run][1]:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1478469497282105344/artifacts/e2e-agnostic/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-5b44c85bd4-qgwdw_cluster-version-operator.log | sed -n 's/.* \([^ ]*[.]go:[0-9]*\)].*/\1/p' | sort | uniq -c | sort -n | tail -n7
    288 cvo.go:514
    288 cvo.go:516
    352 apps.go:38
   1512 sync_worker.go:393
   1558 task_graph.go:477
  14548 sync_worker.go:771
  14551 sync_worker.go:759
```

We still see a handful of the `extraLongThrottleLatency` messages:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1478469497282105344/artifacts/e2e-agnostic/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-5b44c85bd4-qgwdw_cluster-version-operator.log | grep 'Waited for' | head -n3
I0104 21:41:18.122651       1 request.go:665] Waited for 1.006875105s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:6443/api/v1/namespaces/openshift-cluster-samples-operator
I0104 21:41:19.148129       1 request.go:665] Waited for 1.497948234s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:6443/apis/operators.coreos.com/v1/namespaces/openshift-monitoring/operatorgroups/openshift-cluster-monitoring
I0104 21:41:20.172157       1 request.go:665] Waited for 1.370806868s due to client-side throttling, not priority and fairness, request: GET:https://127.0.0.1:6443/apis/operator.openshift.io/v1/csisnapshotcontrollers/cluster

$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1478469497282105344/artifacts/e2e-agnostic/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-5b44c85bd4-qgwdw_cluster-version-operator.log | grep 'Waited for' | wc -l
26
```

but that's acceptably quiet.

/hold cancel

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1478469497282105344

**Comment by wking** [comment]
Hrmm, [update job][1] seems to be wedged on:

```
level=fatal msg=failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": api.ci.azure.devcluster.openshift.com CNAME record already exists in ci-op-twz79rwz-7ee27 and might be in use by another cluster, please remove it to continue
```

I could probably bump the commit or some such to get a new job hash and avoid that issue, but I'm personally confident enough in this change and the other green presubmits to just override the job:

/override ci/prow/e2e-agnostic-upgrade

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/721/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic-upgrade/1478674373081567232#1:build-log.txt%3A67

**Comment by sdodson** [comment]
/cherry-pick release-4.9
Probably conflicts, who knows.

**Pull request merged**
